lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.h

```diff
@@ -147,20 +147,33 @@ class ProxyRenderDelegate : public MHWRender::MPxSubSceneOverride
     // USD & Hydra Objects
     HdEngine                _engine;                    //!< Hydra engine responsible for running synchronization between scene delegate and VP2RenderDelegate
     HdTaskSharedPtrVector   _dummyTasks;                //!< Dummy task to bootstrap data preparation inside Hydra engine
-    UsdStageRefPtr          _usdStage;                  //!< USD stage pointer
+    UsdStageRefPtr          _usdStage;              //!< USD stage pointer
 
     // Do not re-order these 4 member variables relative to one another. The order they are declared here is the reverse order of deletion, and the deletion order matters!
     std::unique_ptr<HdRenderDelegate>  _renderDelegate; //!< VP2RenderDelegate
     std::unique_ptr<HdRenderIndex> _renderIndex;        //!< Flattened representation of client scene graph
     std::unique_ptr<HdxTaskController> _taskController; //!< Task controller necessary for execution with hydra engine (we don't really need it, but there doesn't seem to be a way to get synchronization running without it)
     std::unique_ptr<UsdImagingDelegate> _sceneDelegate; //!< USD scene delegate
 
-    size_t              _excludePrimPathsVersion{ 0 }; //!< Last version of exluded prims used during render index populate
-
-    bool                _isPopulated{ false };      //!< If false, scene delegate wasn't populated yet within render index
-    bool                _selectionChanged{ false }; //!< Whether there is any selection change or not
-    bool                _isProxySelected{ false };  //!< Whether the proxy shape is selected
-    MColor              _wireframeColor;            //!< Wireframe color assigned to the proxy shape
+    // We need to track when output data on the proxy shape changes. For simple numeric types we cache the last value
+    // read from _proxyShape. For complicated types we keep a version number of the last value we read to make
+    // comparisons fast.
+    class ProxyShapeData
```

Comment:
The class name is probably a little bit confusing. How about ProxyShapeVersion?

Reply:
I'm thinking about moving the proxyShape pointer, the _usdStage and anything else we get directly from the proxyShape into this class. What do you think about that idea?

Reply:
I agree.